File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Feedback/FeedbackRecord.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using Microsoft.Bot.Schema;

namespace Microsoft.Bot.Builder.Solutions.Feedback
{
    [Obsolete("This type is being deprecated. It's moved to the assembly Microsoft.Bot.Solutions. Please refer to https://aka.ms/botframework-solutions/releases/0_8", false)]
    public class FeedbackRecord
    {
        /// <summary>
        /// Gets or sets the activity for which feedback was requested.
        /// </summary>
        /// <value>
        /// The activity for which feedback was requested.
        /// </value>
        public Activity Request { get; set; }

        /// <summary>
        /// Gets or sets feedback value submitted by user.
        /// </summary>
        /// <value>
        /// Feedback value submitted by user.
        /// </value>
        public string Feedback { get; set; }

        /// <summary>
        /// Gets or sets free-form comment submitted by user.
        /// </summary>
        /// <value>
        /// Free-form comment submitted by user.
        /// </value>
        public string Comment { get; set; }

        /// <summary>
        /// Gets or sets tag for categorizing feedback.
        /// </summary>
        /// <value>
        /// Tag for categorizing feedback.
        /// </value>
        public string Tag { get; set; }
    }
}

File: sdk/typescript/libraries/bot-solutions/src/feedback/feedbackRecord.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { Activity } from 'botframework-schema';

export class FeedbackRecord {
    /**
     * Gets or sets the activity for which feedback was requested.
     * The activity for which feedback was requested.
     */
    public request?: Activity;

    /**
     * Gets or sets feedback value submitted by user.
     * Feedback value submitted by user.
     */
    public feedback?: string;

    /**
     * Gets or sets free-form comment submitted by user.
     * Free-form comment submitted by user.
     */
    public comment?: string;

    /**
     * Gets or sets tag for categorizing feedback.
     * Tag for categorizing feedback.
     */
    public tag?: string;
}
